### PR TITLE
feat(timeseries): support hourly/gridded products

### PR DESCRIPTION
**Pull request by ocehugo**
1.Add support for categories to hourly/gridded products.
2. Add tests for aggregated/hourly/gridded products pickup and
path resolving.

### CODE REVIEW

File: test_aodndata/moorings/test_mooringsProductsHandler.py [hunk 1]
```diff
@@ -25,33 +28,100 @@
...
+            testfile, {
+                'site_code': 'NRSROT',
+                'source': 'mooring',
+                'featureType': 'timeSeriesProfile'
+            })
+        dest_dir, dest_filename = os.path.split(
+            MooringsProductsHandler.dest_path(testfile))
+        self.assertEqual(dest_dir, expected_prefix)
+        self.assertEqual(dest_filename, filename)
+
```
Comment (mhidas) [inline test_aodndata/moorings/test_mooringsProductsHandler.py]: Unfortunately ABOS have requested a different path structure, so we need separate test cases (and additional logic in the classifier) for them. The difference is they don't include the site code in the path, so it's just `IMOS/ABOS/<sub-facility>/aggregated_timeseries/` etc...
Here are some test cases, with their correct dest paths:
```
IMOS/ABOS/DA/aggregated_timeseries/IMOS_ABOS-DA_TZ_20150519_EAC4700_FV01_TEMP-aggregated-timeseries_END-20180422_C-20191216.nc
IMOS/ABOS/SOTS/hourly_timeseries/IMOS_ABOS-SOTS_BSTUZ_20100912_SOTS_FV02_hourly-timeseries_END-20190322_C-20190819.nc
IMOS/ABOS/DA/gridded_timeseries/IMOS_ABOS-DA_TZ_20150519_EAC4700_FV02_TEMP-gridded-timeseries_END-20180422_C-20191216.nc
```

File: test_aodndata/moorings/test_mooringsProductsHandler.py [hunk 2]
```diff
@@ -25,33 +28,100 @@
 features = json.loads(TEST_GETFEATURE_JSON)['features']
 INPUT_FILE_COLLECTION = PipelineFileCollection()
 for f in features:
-    pf = PipelineFile(
-        os.path.join(TEST_ROOT, os.path.basename(f['properties']['url'])),
-        dest_path=f['properties']['url']
-    )
+    pf = PipelineFile(os.path.join(TEST_ROOT,
+                                   os.path.basename(f['properties']['url'])),
+                      dest_path=f['properties']['url'])
```
[the remark this answers is not in the document]
Comment (ocehugo) [reply test_aodndata/moorings/test_mooringsProductsHandler.py]: I think this is pep8 - line width limits - but will check

File: test_aodndata/moorings/test_mooringsProductsHandler.py [hunk 3]
```diff
@@ -25,33 +28,100 @@
...
-                               responses=[TEST_GETCAPABILITIES_RESPONSE, TEST_GETCAPABILITIES_RESPONSE,
-                                          TEST_GETFEATURE_RESPONSE]
-                               )
+        httpretty.register_uri(
+            httpretty.GET,
+            self.config.pipeline_config['global']['wfs_url'],
+            responses=[
+                TEST_GETCAPABILITIES_RESPONSE, TEST_GETCAPABILITIES_RESPONSE,
+                TEST_GETFEATURE_RESPONSE
+            ])
```
[the remark this answers is not in the document]
Comment (ocehugo) [reply test_aodndata/moorings/test_mooringsProductsHandler.py]: As above

File: test_aodndata/moorings/test_mooringsProductsHandler.py [hunk 4]
```diff
@@ -25,33 +28,100 @@
...
 
-        upload_broker = get_storage_broker(self.config.pipeline_config['global']['upload_uri'])
+        upload_broker = get_storage_broker(
+            self.config.pipeline_config['global']['upload_uri'])
         upload_broker.upload(INPUT_FILE_COLLECTION)
 
         handler = self.run_handler(GOOD_MANIFEST)
         self.assertEqual(len(handler.excluded_files), 1)
 
+    def test_aodn_moorings_aggregated(self):
```
[the remark this answers is not in the document]
Comment (ocehugo) [reply test_aodndata/moorings/test_mooringsProductsHandler.py]: I don't get it. The filename is testproducthandler and I'm using my the class name handler.

Also, I put aodn moorings to disambiguate -  a gridded product already exist

File: test_aodndata/moorings/test_mooringsProductsHandler.py [hunk 5]
```diff
@@ -25,33 +28,100 @@
...
+        expected_prefix = 'IMOS/ANMN/QLD/GBRLSL/aggregated_timeseries'
+        filename = 'IMOS_ANMN-QLD_BZ_20121103_GBRLSL_FV01_CPHL-aggregated-timeseries_END-20140522_C-20191120.nc'
+        testfile = os.path.join(self.tempdir, filename)
+        make_test_file(
+            testfile, {
+                'site_code': 'GBRLSL',
+                'source': 'mooring',
+                'featureType': 'timeSeries'
+            })
```
[the remark this answers is not in the document]
Comment (ocehugo) [reply test_aodndata/moorings/test_mooringsProductsHandler.py]: The checkhere don't need the variable name so I didn't included

**Comment by ocehugo** [comment]
As above - I follow name conventions in the file - handler is different than classifier so I use handler classes.

File: test_aodndata/moorings/test_mooringsProductsHandler.py [hunk 6]
```diff
@@ -25,33 +28,100 @@
...
+        dest_dir, dest_filename = os.path.split(
+            MooringsProductsHandler.dest_path(testfile))
```
[the remark this answers is not in the document]
Comment (ocehugo) [reply test_aodndata/moorings/test_mooringsProductsHandler.py]: As above - I don't get it - Im follow name conventions in the file - handler is different than classifier so I use handler classes

File: test_aodndata/moorings/test_mooringsProductsHandler.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ocehugo) [reply test_aodndata/moorings/test_mooringsProductsHandler.py]: Yes - pep8/pycodestyle. Are we following pep8!?

**Comment by ocehugo** [comment]
@mhidas, travis is also beeping regarding f-strings which are not supported in 3.5 - I will change that.

File: test_aodndata/moorings/test_mooringsProductsHandler.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mhidas) [reply test_aodndata/moorings/test_mooringsProductsHandler.py]: My code here is 75 characters wide (your update is 3 chars _longer_)

Generally yes, we do try to follow PEP8, but we're not limiting line width to 79 characters.

File: test_aodndata/moorings/test_mooringsProductsHandler.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mhidas) [reply test_aodndata/moorings/test_mooringsProductsHandler.py]: As I said, these lines are about to be removed in another PR, so changing them here will just cause merge conflicts.

File: test_aodndata/moorings/test_mooringsProductsHandler.py [hunk 7]
```diff
@@ -25,33 +28,100 @@
...
 
 
 class TestMooringsProductsHandler(HandlerTestCase):
     def setUp(self):
+        self.tempdir = mkdtemp()
         self.handler_class = MooringsProductsHandler
         super(TestMooringsProductsHandler, self).setUp()
 
+    def tearDown(self):
+        shutil.rmtree(self.tempdir)
```
[the remark this answers is not in the document]
Comment (ocehugo) [reply test_aodndata/moorings/test_mooringsProductsHandler.py]: I inadvertently click resolved. The method is required to be able to run the tests

**Comment by ocehugo** [comment]
check it out @mhidas

**Comment by ocehugo** [comment]
sorry about the early call - Forgot to check conflicts and the travis/jenkins builds were still kicking in.

**Comment by ocehugo** [comment]
all good apparently - codecoverage is complaining about not hitting some  exceptions - but this is partially a problem with the test suite not hitting error cases and because of deep inheritance.
@mhidas

**Comment by mhidas** [comment]
@ocehugo I've just added a couple of commits to allow the handler to accept product files created elsewhere and publish them. Assuming Travis is still happy, I will merge this and test in RC.

**Comment by ocehugo** [comment]
I just don't get why you add the name_fields variable since it's not used !
all else good & good catch on the docstrings update.

**Comment by mhidas** [comment]
> I just don't get why you add the name_fields variable since it's not used !

I'm removing it, not adding (see [here](https://github.com/aodn/python-aodndata/pull/188/commits/0c5dc99882946547b8f7ebf86632c68119a30ade#diff-b03823a90d19c1c432001f900d93f24aL58))

**Pull request merged**